### PR TITLE
swapon: use a loopback device

### PR DESCRIPTION
**Pull request by goneri**
btrfs does not support swapfile. With a loopback device, we bypass the
limitation.

### CODE REVIEW

File: tools/test-setup.sh [hunk 1]
```diff
@@ -4,8 +4,9 @@ set -euxo pipefail
 
 sudo dd if=/dev/zero of=/swap.img bs=1024 count=1048576
 sudo chmod 600 /swap.img
-sudo mkswap /swap.img
-sudo swapon /swap.img
+sudo losetup -f /swap.img
+sudo mkswap $(sudo losetup --associated swap.img|sed 's,:.*,,')
```
Comment (apatard) [inline tools/test-setup.sh:8]: hm. I guess you're missing a '/' here, leading to the error:
```
mkswap: error: Nowhere to set up swap on?
```

(btw, thanks for looking at this, I failed to find time to work on this swap thing so far)

Comment (ssbarnea) [reply tools/test-setup.sh:8]: Is this going to work even if we run on systems w/o btrfs? I really want to keep test-setup.sh as "portable" as possible.

Comment (apatard) [reply tools/test-setup.sh:8]: I've not tested it but I guess that it should work on local filesystems.

Comment (goneri) [reply tools/test-setup.sh]: Do you really need this step?

Comment (apatard) [reply tools/test-setup.sh]: the tests are creating one or several VMs and it's asking some memory. iirc, the test suite got issues in the past due to this, so adding a swap file helps.

File: tools/tox-py39/pre.yml [hunk 2]
```diff
@@ -1,8 +1,17 @@
 ---
 - hosts: all
   tasks:
-    - name: Ensure python3.9 is present
+    - name: Ensure python3.9 is present (Debian/Ubuntu)
       become: true
       package:
-        name: python
+        name: python3i9-devel
```
Comment (apatard) [inline tools/tox-py39/pre.yml:7]: python39-devel maybe ?

File: tools/tox-py39/pre.yml [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
         state: present
+      when: ansible_os_family != "RedHat"
+    - name: Ensure python3.8 is present (Red Hat)
+      become: true
+      package:
+        name:
+          - python3.9
```
Comment (apatard) [inline tools/tox-py39/pre.yml:14]: either 3.9 is wrong here or the comment about 3.8 is.

File: tools/tox-py39/pre.yml (hunk 2 above)
Comment (goneri) [reply tools/tox-py39/pre.yml:7]: ahah :-)

**Comment by apatard** [comment]
recheck

**Review by ssbarnea — APPROVED** [review]

**Comment by apatard** [comment]
recheck

**Comment by ssbarnea** [comment]
Zuul fails with error below, I think we need to address this first:

> Error while creating domain: Error saving the server: Call to virDomainDefineXML failed: invalid argument: could not get preferred machine for /usr/bin/qemu-system-x86_64 type=kvm"

**Comment by apatard** [comment]
> Zuul fails with error below, I think we need to address this first:
> 
> > Error while creating domain: Error saving the server: Call to virDomainDefineXML failed: invalid argument: could not get preferred machine for /usr/bin/qemu-system-x86_64 type=kvm"

As said previously, this particular issue is supposed to be fixed by https://github.com/ansible-community/molecule-vagrant/pull/158 but it needs this PR otherwise zuul will likely fail badly at least on fedora due to memory issues.

**Comment by ssbarnea** [comment]
OK, lets YOLO it. ;)

**Pull request merged**